src/condition/cmdi_detector.cpp

```diff
@@ -0,0 +1,263 @@
+    // contained within the std::string, we ensure a new one won't be
+    // allocated once the string is modified.
+    cow_string executable_lc{executable};
+    if (transformer::lowercase::transform(executable_lc)) {
+        executable = static_cast<std::string_view>(executable_lc);
+    }
+
+    auto shell_it = known_shells.find(basename(executable));
+    if (shell_it != known_shells.end()) {
+        // We've found that the current exec command is attempting to run a
+        // a shell. The shell binary itself might be injected, but also the
+        // shell command. So we need to identify the command
+        std::size_t i = 1;
+        if (!shell_it->second.empty()) {
+            // Most shells allow specifying a command with -c
+            for (; i < object_size(exec_args); ++i) {
+                auto opt = trim_whitespaces(object_at(exec_args, i));
+                if (!opt.empty() && opt[0] == '-' &&
+                    opt.find(shell_it->second) != std::string_view::npos) {
+                    // We've found the -c option, we can now break, if it isn't found
+                    // i will reach the end of the array
+                    break;
+                }
+            }
+        }
+        for (; i < object_size(exec_args); ++i) {
+            auto arg = trim_whitespaces(object_at(exec_args, i));
+            if (arg.empty() || arg.starts_with('-') || arg.starts_with('+')) {
+                continue;
+            }
+
+            // Once the first non-option argument is reached, it must be the
+            // shell command
+            return arg;
+        }
+    }
+    return {};
+}
+
+std::optional<shi_result> cmdi_impl(const ddwaf_object &exec_args,
+    std::vector<shell_token> &resource_tokens, const ddwaf_object &params,
+    const exclusion::object_set_ref &objects_excluded, const object_limits &limits,
+    ddwaf::timer &deadline)
+{
+    const std::string_view executable = trim_whitespaces(object_at(exec_args, 0));
+    if (executable.empty()) {
+        return {};
+    }
+
+    auto shell_command = find_shell_command(executable, exec_args);
+
+    object::kv_iterator it(&params, {}, objects_excluded, limits);
+    for (; it; ++it) {
+        if (deadline.expired()) {
+            throw ddwaf::timeout_exception();
+        }
+
+        const ddwaf_object &param = *(*it);
+        if (param.type != DDWAF_OBJ_STRING) {
+            continue;
+        }
+
+        // First check if the entire executable was injected
+        std::string_view value{param.stringValue, static_cast<std::size_t>(param.nbEntries)};
+        value = trim_whitespaces(value);
+        if (executable == value) {
+            // When the full binary has been injected, we consider it an exploit
+            // although bear in mind that this can also be a vulnerable-by-design
+            // application, leading to a false positive
+            return {{std::string(value), it.get_current_path()}};
```

Comment:
seems a bit strict, I mean if the program happens to execute sh, any parameter with the value 'sh' will trigger this, right?

Reply:
Yes, I  agree, I already mentioned it to @Taiki-San but he didn't seem concerned...

Reply:
I'm not definitive on this point. If we're concerned, we can keep it in reserve for the Webshell detection

Reply:
Alright, restricting this for now to paths with more than one component (i.e. `/` or `\`).